### PR TITLE
Add support for archiving liked posts

### PR DESCRIPTION
**Pull request by Sorixelle**
this pr defines a new `--liked` flag on both `cohost-archive` and `cohost2json`. when provided, autost will include the liked posts for the requested page in the archive.

notes:
- archiving liked posts only works for the logged in project. `cohost-archive` will only try to fetch liked posts for that project, and `cohost2json` will skip archiving liked posts if they were requested but no `COHOST_COOKIE` is provided
- liked posts likely aren't authored by the `self_author` for the site, so they won't end up anywhere other than `skipped_other.html`  in the rendered site without extra configuration

### CODE REVIEW

File: src/cohost.rs [hunk 1]
```diff
@@ -183,6 +183,32 @@ pub struct FollowedProject {
     pub handle: String,
 }
 
+#[derive(Debug, Deserialize)]
+#[allow(non_snake_case)]
+pub struct LikedPostsState {
+    #[serde(rename = "liked-posts-feed")]
+    pub liked_posts_feed: LikedPostsFeed,
```
Comment (delan) [inline src/cohost.rs:190]: autost’s convention is to use original field names, could you please rename?

**Review by delan — CHANGES_REQUESTED** [review]
this looks great!! one minor change…

File: src/command/cohost_archive.rs [hunk 2]
```diff
@@ -107,14 +110,16 @@ pub async fn main(args: CohostArchive) -> eyre::Result<()> {
         .collect::<Vec<_>>();
```
Comment (delan) [inline src/command/cohost_archive.rs:111]: if the user runs this with both `--liked` and specific projects, what should we do if none of them are the project you’re logged in as? maybe `bail!()` with an error?

File: src/cohost.rs (hunk 1 above)
Comment (delan) [reply src/cohost.rs:190]: ah. i see

Comment (delan) [reply src/cohost.rs:190]: ah. i see. never mind ^w^;;

Comment (delan) [reply src/cohost.rs:190]: image

File: src/command/cohost_archive.rs (hunk 2 above)
Comment (Sorixelle) [reply src/command/cohost_archive.rs:111]: part of me wants to say the flag would simply be a no-op in this case - given the help text says "your" posts specifically i'm not sure this warrants bailing, but i'm open to other ideas. maybe a warning?

File: src/cohost.rs (hunk 1 above)
Comment (Sorixelle) [reply src/cohost.rs:190]: no can do - original field has hyphens in it

File: src/command/cohost_archive.rs (hunk 2 above)
Comment (delan) [reply src/command/cohost_archive.rs:111]: yea, warning works

File: src/command/cohost_archive.rs [hunk 3]
```diff
@@ -110,7 +110,10 @@ pub async fn main(args: CohostArchive) -> eyre::Result<()> {
...
 
     if args.liked && !project_names.contains(&logged_in_project.handle) {
-        warn!("requested liked posts, but not the logged in project - skipping liked posts");
+        warn!(
+            "requested liked posts, but not the logged in project (@{}) - skipping liked posts",
```
Comment (delan) [inline src/command/cohost_archive.rs:114]:
```suggestion
            "you requested liked chosts, but not your own logged in project (@{}); skipping liked chosts",
```

**Review by delan — APPROVED** [review]
thanks!! minor rewording and we’re good to go :)

**Review by delan — APPROVED** [review]
ao!! wrrrrao!!

**Pull request merged**